paymentsheet/src/main/java/com/stripe/android/paymentsheet/DeferredIntentValidator.kt

```diff
@@ -0,0 +1,81 @@
+package com.stripe.android.paymentsheet
+
+import com.stripe.android.model.DeferredIntentParams
+import com.stripe.android.model.ElementsSessionParams
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.PaymentIntent.CaptureMethod
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.paymentsheet.repositories.toElementsSessionParams
+
+@OptIn(ExperimentalPaymentSheetDecouplingApi::class)
+internal object DeferredIntentValidator {
+
+    /**
+     * Validates that the created [StripeIntent] matches the [PaymentSheet.IntentConfiguration] that
+     * was provided to [PaymentSheet].
+     */
+    fun validate(
+        stripeIntent: StripeIntent,
+        intentConfiguration: PaymentSheet.IntentConfiguration,
+        isFlowController: Boolean,
```

Comment:
Rather than having a boolean here, could we inject custom validation rule callback, and implement that in `FlowController`?

Reply:
Injecting them now!